### DISCUSSION TITLE
JSON library | invalid utf-8 byte at index

### DISCUSSION THREAD

**Discussion by SkYMaaN** [nlohmann/json#2455] [Q&A] [answered]
When i try output json array i get exception like it:
 std::string sn(3, '\0');
(std::snprintf)(&sn[0], sn.size(), "%.2X", byte);
JSON_THROW(type_error::create(316, "invalid UTF-8 byte at index " + std::to_string(i) + ": 0x" + sn));
How i can resolve it?

**Comment by nlohmann** [comment, marked as answer]
What is your input? Can you make sure it is really UTF-8 encoded?

**Reply by SkYMaaN** [reply]
Input data it's data from each cell in Excel sheet ( .xlsx ). My program receive this data and cast it to wstring after that - put in JSON array.

**Reply by nlohmann** [reply]
Data in `std::wstring` is not UTF-8 encoded, and this library only supports UTF-8. How exactly do you call the library's parser?

**Reply by SkYMaaN** [reply]
Am I assigning data without calling the library parser
if (col == 3 && row > 5) //excel col and row
                                {
                                    const wchar_t* s = sheet->readStr(row, col);  //read pointer
                                    wstring wstr(s);  //cast to wstring
                                    j["р.Дніпро(річкова_вода)"] = wstr;  //put into JSON
                                }

**Reply by nlohmann** [reply]
I see. Then I'm afraid you would either need to convert your input to UTF-8. Or you set the error handler in the [`dump`](https://nlohmann.github.io/json/api/basic_json/dump/) function to `ignore` to not have an error at output.

**Reply by SkYMaaN** [reply]
Thanks you! I solved this problem.

**Reply by Tyoma7Tmp** [reply]
Please, tell me how you solved this problem.
